sdk/core/Azure.Core/tests/DynamicData/DynamicJsonTests.cs

```diff
@@ -844,6 +844,100 @@ public void CanDifferentiateBetweenNullAndAbsent()
+        {
+            dynamic json = BinaryData.FromString("""{ "data" : null }""").ToDynamicFromJson(DynamicCaseMapping.PascalToCamel);
+
+            var value = new
+            {
+                foo = 1
+            };
+
+            BinaryData serializedValue = BinaryData.FromObjectAsJson(value);
+
+            // existing property
+            json.Data = serializedValue;
+
+            // new property
+            json.NewData = serializedValue;
+
+            // To get the value back from BinaryData property, we must first obtain it as BinaryData.
+            BinaryData data = json.Data;
+            BinaryData newData = json.NewData;
+
+            // And now you can use dynamic data to read the values.
+            dynamic first = data.ToDynamicFromJson(DynamicCaseMapping.PascalToCamel);
+            dynamic second = newData.ToDynamicFromJson(DynamicCaseMapping.PascalToCamel);
+
+            Assert.AreEqual(1, (int)first.Foo);
+            Assert.AreEqual(1, (int)second.Foo);
+        }
+
+        [Test]
+        public void CanSetBinaryDataValueModelType()
+        {
+            dynamic json = BinaryData.FromString("""{ "data" : null }""").ToDynamicFromJson(DynamicCaseMapping.PascalToCamel);
+
+            SampleModel model = new SampleModel() { Number = 1, Message = "hi" };
+
+            BinaryData serializedValue = BinaryData.FromObjectAsJson(model);
+
+            // existing property
+            json.Data = serializedValue;
+
+            // new property
+            json.NewData = serializedValue;
+
+            // To get the value back from BinaryData property, we must first obtain it as BinaryData.
+            BinaryData data = json.Data;
+            BinaryData newData = json.NewData;
+
+            SampleModel first = data.ToObjectFromJson<SampleModel>();
+            SampleModel second = newData.ToObjectFromJson<SampleModel>();
+
+            Assert.AreEqual(1, first.Number);
+            Assert.AreEqual(1, second.Number);
+        }
+
+        [Test]
+        public void CanSerializeModelWithBinaryDataProperty()
+        {
+            string json = """{ "foo": null }""";
+            dynamic value = BinaryData.FromString(json).ToDynamicFromJson();
+
+            BinaryData b = BinaryData.FromString("b");
+
+            BinaryDataModel model = new BinaryDataModel()
+            {
+                StringProperty = "a",
+                BinaryDataProperty = b
+            };
+
+            // Existing property
+            value.Foo = model;
+
+            // Added property
+            value.Bar = model;
+
+            Assert.AreEqual("a", (string)value.Foo.StringProperty);
+            CollectionAssert.AreEqual(b.ToArray(), ((BinaryData)value.Foo.BinaryDataProperty).ToArray());
+
+            Assert.AreEqual("a", (string)value.Bar.StringProperty);
+            CollectionAssert.AreEqual(b.ToArray(), ((BinaryData)value.Bar.BinaryDataProperty).ToArray());
+        }
+
```

Comment:
Could you also add tests for the invalid base64 format cases? Wrong chars, lack of padding, etc.? You could double one of those tests for base64url so the failure check will itself fail should we support that in the future. E.g.,

Base64: DEAD+BEEF/==
Base64url: DEAD-BEEF_
Hex: 0c40033f110417